demo/dlb-init.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+ENABLED_DLB_PF_PCIIDS=${ENABLED_DLB_PF_PCIIDS:-270b 2710 2714}
+NUMVFS=${NUMVFS:-4}
+
+bind() {
+    BSF=$1
+    DRIVER=$2
+
+    LOOP=0
+    while [ "$LOOP" -lt  5 ]
+    do
+        if echo -n 2> /dev/null "$BSF" > /sys/bus/pci/drivers/"$DRIVER"/bind; then
```

Comment:
Bind and unbind is basically copied from some work of qat. I don't remember the rerpo.
I have no idea on this level code, so I just copied.

If you have another idea, just let me know.
I think,,, I am not able to make my own code for 'binding' and 'unbinding'.
The same thing goes for the comment below.

Reply:
I don't think we need this. Here is how it's done currently in DLB CI and it works just fine:
```
# Enable VF for the first DLB device
device="0000:$(lspci -d :2710 | head -1 | cut -f1 -d ' ')"
sysfs="/sys/bus/pci/devices/$device"
echo 1 > "$sysfs/sriov_numvfs"

# configure VFs
num=0
for vf in $(lspci -d :2711 | cut -f1 -d' '); do
  echo -e "\n== Configuring device $num $vf ==\n"
  # unbind VF before configuring it
  dpdk-devbind.py --unbind "0000:$vf"

  # configure
  path="$sysfs/vf$((num))_resources"
  echo 2048 > "$path/num_atomic_inflights"
  echo 2048 > "$path/num_dir_credits"
  echo 8 > "$path/num_dir_ports"
  echo 2048 > "$path/num_hist_list_entries"
  echo 8192 > "$path/num_ldb_credits"
  echo 4 > "$path/num_ldb_ports"
  echo 32 > "$path/num_ldb_queues"
  echo 32 > "$path/num_sched_domains"
  echo 2 > "$path/num_sn0_slots"
  echo 2 > "$path/num_sn1_slots"

  num=`expr $num + 1`

  # bind VF back
  dpdk-devbind.py --bind dlb2 "0000:$vf"
done
```

Reply:
We are removing dpdk-devbind to remove python and dpdk dependencies from our initimage.! It's the reason why I copied some bind/unbind func that substitutes dpdk-devbind.py.

Reply:
https://github.com/intel/qatlib/blob/main/quickassist/utilities/service/qat_init.sh.in

Reply:
@bart0sh `dpdk-devbind` pulls in python and DPDK so I proposed to do this without them.

Reply:
> We are trying to make dpdk-devbind not have dependencies on python and dpdk.! It's the reason why I copied some bind/unbind func that substitutes dpdk-devbind.py

Not sure I understand. Do you mean that '2>/dev/null' is used because you're not using dpdk-devbind?

Reply:
no... I mean..
Mikko said it's better not to use dpdk-devbind but just implement 'bind' and 'unbind' funcs, because we do not need 'python' and 'DPDK' inside the initimage if we implment them.
So, he gave [this link from qatlib](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1088) and I just copied from there.
And you pointed out some parts of the function, and I commented above that I do not know what they mean honestly...

So, if it does not make sense, please just let me know how it should be...
and maybe we need to make a PR in qatlib also if this does not make sense.

Reply:
Okay, I think about this matter, maybe, they have their own reason why they do not want to see the error if it fails. But, for us, it would be better to see. Let me remove them. But, for the looping 5 times matter, I want to know why we need to have any other specific number if it has to be changed.

Reply:
The question was about hiding error output. dpdk-devbind is irrelevant here, I believe.

Reply:
yes, but what I and Mikko wanted to say was... when I said I do not know what to do, you gave the code 'dpdk-devbind'. So, we were explaining that goal is to remove the dependency on dpdk and python from our initimage by just putting the functions bind/unbind.

Reply:
The code I've shown demonstrates two things:
- error output is not hidden
- there is not need to bind VF more than one time

The fact that code used dpkg-devbind is irrelevant. I don't know why you become so consentrated on it.